### PR TITLE
[TorchToLinalg] Add support for circular padding ops

### LINKED ISSUE #24585: [Bug] Pad with mode="wrap" causes compiler to hang (infinite loop)
### What happened?

The `Pad` op with `mode="wrap"` causes `iree-compile` to hang indefinitely (tested >120 seconds on a trivial 4×4 input). The compiler appears to enter an infinite loop. Other pad modes (`constant`, `reflect`, `edge`) compile and run correctly on the same input.

ONNX Runtime handles wrap padding correctly.

This also affects PyTorch models that use `torch.nn.functional.pad(mode='circular')`, which exports as `Pad(mode='wrap')` in ONNX.

### Other pad modes (control, same input and pads)

| Mode | Result |
|---|---|
| `constant` | Compiles and runs correctly |
| `reflect` | Compiles and runs correctly |
| `edge` | Compiles and runs correctly |
| `wrap` | **Hangs indefinitely** |

### Impact

Wrap padding is used for circular/periodic padding in signal processing and models that require toroidal boundary conditions. In PyTorch, `torch.nn.functional.pad(mode='circular')` exports as ONNX `Pad(mode='wrap')`, so any PyTorch model using circular padding will trigger this hang.

### Expected behavior

Compilation should either succeed and produce correct output, or fail with a clear error message — not hang indefinitely.

### Environment

- IREE compiler: `3.12.0rc20260515`
- Target: `llvm-cpu` (generic)
- ONNX opset: 19

### Steps to reproduce

```python
import numpy as np
import onnx
from onnx import helper, TensorProto, numpy_helper

X = helper.make_tensor_value_info("X", TensorProto.FLOAT, [1, 1, 4, 4])
Y = helper.make_tensor_value_info("Y", TensorProto.FLOAT, None)
pads = numpy_helper.from_array(np.array([0,0,1,1,0,0,1,1], dtype=np.int64), "pads")
nodes = [helper.make_node("Pad", ["X", "pads"], ["Y"], mode="wrap")]
graph = helper.make_graph(nodes, "test", [X], [Y], initializer=[pads])
model = helper.make_model(graph, opset_imports=[helper.make_opsetid("", 17)])
onnx.save(model, "/tmp/pad_wrap.onnx")
```

```bash
iree-import-onnx /tmp/pad_wrap.onnx -o /tmp/pad_wrap.mlir  # OK
iree-compile --iree-hal-target-backends=llvm-cpu --iree-llvmcpu-target-cpu=generic \
  /tmp/pad_wrap.mlir -o /tmp/pad_wrap.vmfb
# Hangs indefinitely — no output, no error, no progress
```

### What component(s) does this issue relate to?

_No response_

### Version information

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AkunCube**
This PR adds support for `torch.circular_pad{1|2|3}d` ops in TorchToLinalg.

- Introduces `AtenCircularPad{1|2|3}d` ODS definitions in the Torch dialect
- Implements lowering to `linalg.generic` using modulo-based circular indexing
- Refactors pad mode dispatch in `DecomposeComplexOps` with a macro to reduce duplication across reflect/replicate/circular modes

Fixes https://github.com/iree-org/iree/issues/24585

### CODE REVIEW

**Comment by AkunCube** [comment]
Would be great if someone could review this — @frederik-h / @vinitdeodhar

**Comment by AkunCube** [comment]
Would be great if someone could review this — @penguin-wwy

File: include/torch-mlir/Dialect/Torch/IR/GeneratedTorchOps.td [hunk 1]
```diff
@@ -10556,6 +10530,78 @@ def Torch_AtenReplicationPad3dOp : Torch_Op<"aten.replication_pad3d", [
   }];
 }
 
+def Torch_AtenCircularPad1dOp : Torch_Op<"aten.circular_pad1d", [
```
Comment (vinitdeodhar) [inline include/torch-mlir/Dialect/Torch/IR/GeneratedTorchOps.td:10533]: This file should be generated from torch_ods_gen. Please refer to: https://github.com/llvm/torch-mlir/blob/main/docs/add_ops.md
